File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Commands/Attachments/Put.cs
using System.IO;

using Raven.Abstractions.Data;
using Raven.Client.Document;
using Raven.Json.Linq;

namespace Raven.Documentation.Samples.ClientApi.Commands.Attachments
{
	public class Put
	{
		private interface IFoo
		{
			#region put_1
			void PutAttachment(string key, Etag etag, Stream data, RavenJObject metadata);
			#endregion
		}

		public Put()
		{
			using (var store = new DocumentStore())
			{
				#region put_2
				using (var file = File.Open("sea.png", FileMode.Open))
				{
					store
						.DatabaseCommands
						.PutAttachment("albums/holidays/sea.jpg", null, file, new RavenJObject
							                                           {
								                                           { "Description", "Holidays 2014" }
							                                           });
				}
				#endregion
			}
		}
	}
}

File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/commands/attachments/Put.java
package net.ravendb.clientapi.commands.attachments;

import java.io.FileInputStream;
import java.io.InputStream;

import net.ravendb.abstractions.data.Etag;
import net.ravendb.abstractions.json.linq.RavenJObject;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;


public class Put {

  @SuppressWarnings("unused")
  private interface IFoo {
    //region put_1
    @Deprecated
    public void putAttachment(String key, Etag etag, InputStream data, RavenJObject metadata);
    //endregion
  }

  @SuppressWarnings("deprecation")
  public Put() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      //region put_2
      try (FileInputStream fis = new FileInputStream("sea.png")) {
        RavenJObject metadata = new RavenJObject();
        metadata.add("Description", "Holidays 2014");
        store
        .getDatabaseCommands()
        .putAttachment("albums/holidays/sea.jpg", null, fis, metadata);
      }
      //endregion
    }
  }
}
